py/redrock/zfind.py

```diff
@@ -33,8 +33,23 @@
 
 from .zwarning import ZWarningMask as ZW
 
-
-def _mp_fitz(chi2, target_data, t, nminima, qout, archetype):
+def sort_dict_by_col(d, colname):
+    """Sort a dict of np.ndarrays by one key.
+    Replacement for astropy.Table.sort
+    """
+    if (not colname in d):
+        raise KeyError('Key '+str(colname)+' is not in dictionary')
+    if (type(d[colname]) is not np.ndarray):
+        raise ValueError('Column '+str(colname)+' is not an np.array')
+    idx = d[colname].argsort(0).flatten()
+    for k in d.keys():
+        if (type(d[k]) is not np.ndarray):
+            raise ValueError('Column '+str(k)+' is not an np.array')
```

Comment:
Minor, pedantic: let's do the type checking and raise any exceptions before we starting modifying `d`.  Otherwise a user who catches an exception could find their input array corrupted with half the columns reordered and half not.  It might be even better to cache the reordered columns into a different dict and only update the input in-place after we are completely sure that every column is sortable (though this would use more memory).

Reply:
This is resolved in the version I just pushed out - I check all columns before re-ordering any.